AI #1663: Billed Cost and Effective Cost definitions update (#1678)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/datasets/cost_and_usage/columns/effectivecost.md

# Effective Cost

> Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the [*resources*](#glossary:resource) or [*services*](#glossary:service) used or [*contract commitments*](#glossary:contract-commitment) consumed in a given [*charge period*](#glossary:charge-period). Effective Cost differs from [Billed Cost](#datasets.costandusage.billedcost) when pre-payment or post-payment *charges* are invoiced separately from usage.

Reviewer: Can we use "consumed" and "used" as synonyms here?  Looking to simplify that list of three things: resources, services, and contract commitments.

See my comment on Billed Cost [here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1678/files#r2713583663) for why I suggest striking "pre-payment and post-payment", and why I'm suggesting a net-new glossary entry here.

```suggestion
Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the [*resources*](#glossary:resource), [*services*](#glossary:service), or [*contract commitments*](#glossary:contract-commitment) consumed in a given [*charge period*](#glossary:charge-period). Effective Cost differs from [Billed Cost](#datasets.costandusage.billedcost) when *charges* are [*invoiced*](#glossary:invoice-process) separately from usage.
```

Reviewer 2: Same answer. We've spent a bunch of time wordsmithing here. It's being verbose to ensure that unconsumed/unused portions, just like used/consumed portions, are accounted for (e.g., unused commitments). I would say, a bit more strongly than the above change, that this should remain verbose.

Reviewer: Per Jan 22 Members discussion:

```suggestion
Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the [*resources*](#glossary:resource) used, [*services*](#glossary:service) used, or [*contract commitments*](#glossary:contract-commitment) consumed in a given [*charge period*](#glossary:charge-period). Effective Cost differs from [Billed Cost](#datasets.costandusage.billedcost) when *charges* (both pre-paid and post-paid) are invoiced separately from usage.
```

> Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the [*resources*](#glossary:resource) used, [*services*](#glossary:service) used, or [*contract commitments*](#glossary:contract-commitment) consumed in a given [*charge period*](#glossary:charge-period). Effective Cost differs from [Billed Cost](#datasets.costandusage.billedcost) when *charges* (both pre-paid and post-paid) are invoiced separately from usage.

Reviewer: I would like the "differs from" sentence in Billed Cost and Effective Cost to read exactly the same in both entries.

Billed Cost says:

> Billed Cost differs from Effective Cost when charges are invoiced separately from usage.

Effective Cost says:

> Effective Cost differs from Billed Cost when charges (both pre-paid and post-paid) are invoiced separately from usage.

We should have that parenthetical in both places, or in neither.  Thus, either accept the below suggestion striking it from Effective Cost, or add it back to Billed Cost.

```suggestion
Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the [*resources*](#glossary:resource) used, [*services*](#glossary:service) used, or [*contract commitments*](#glossary:contract-commitment) consumed in a given [*charge period*](#glossary:charge-period). Effective Cost differs from [Billed Cost](#datasets.costandusage.billedcost) when *charges* are invoiced separately from usage.
```

Reviewer 2: I would propose we go the other way and add the missing part to the BilledCost (as described before, the preference was to be more exhaustive in the description to ensure people realize that effective cost is relevant in both pre- and post-paid scenarios).

Reviewer: Sounds good to me!

> Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the [*resources*](#glossary:resource) used, [*services*](#glossary:service) used, or [*contract commitments*](#glossary:contract-commitment) consumed in a given [*charge period*](#glossary:charge-period). Effective Cost differs from [Billed Cost](#datasets.costandusage.billedcost) when *charges* (both pre-paid and post-paid) are invoiced separately from usage.

Reviewer 2: ```suggestion
Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the [*resources*](#glossary:resource) used, [*services*](#glossary:service) used, or [*contract commitments*](#glossary:contract-commitment) recognized in a given [*charge period*](#glossary:charge-period). Effective Cost differs from [Billed Cost](#datasets.costandusage.billedcost) when *charges* (both pre-paid and post-paid) are invoiced separately from usage.
```

> For all *charges*, Effective Cost reflects all applicable pricing adjustments, such as reduced pricing from [*negotiated discounts*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). For usage *charges*, Effective Cost includes the consumed or recognized portion of *Billed Cost* from a related purchase charge. For purchase *charges*, Effective Cost excludes any amounts recognized or will be recognized in related usage *charges* (e.g., usage covered by *commitments*, pre-payments, or marketplace purchases), regardless of when those related *charges* are invoiced.

Reviewer 3: I get what we are trying to do here, but, a common question that practitioners ask, for which AWS does not provide a column, is:  where can I see the net amortized costs?

Removing amortization from the definition confuses things - please add it back, along with some of the other specifics that don't apply to amortization, as we will be making things less clear with this update without it (practitioners will ask where the amortized cost column in FOCUS is and we'll point them to this and it's not clear to me that this is it when reading the definition).

In my opinion: having an amortizated cost column, which has always been EffectiveCost, was a huge win.

Reviewer: Declaration of amortization as an example, rather than "this is amortized data", would be a way forward.

Author: Amortization is mentioned as an example in the updated version. 
Resolving this conversation. Feel free to reopen it if the concern is not adequately addressed.

> For all *charges*, Effective Cost reflects all applicable pricing adjustments, such as reduced pricing from [*negotiated discounts*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). For usage *charges*, Effective Cost includes the consumed or recognized portion of *Billed Cost* from a related purchase charge (e.g. amortized portions of prepayments, draw downs). For purchase *charges*, Effective Cost excludes any amounts recognized or will be recognized in related usage *charges* (e.g., usage covered by *commitments*, pre-payments, or marketplace purchases which draw down based on usage), regardless of when those related *charges* are invoiced.

Reviewer: We're using a mixture of "e.g.," and "such as".  Recommend standardizing.

Is `consumed or recognized` necessary here?  It confuses my understanding to include both terms.

This definition feels like we want to use the term "related" as though it means something specific in FOCUS.  I agree.  If so, perhaps we stand up a glossary entry for "related", which would free us up to use that term as a shorthand.

Do we need `recognized or will be recognized`?  The sentence later specifies `regardless of when those related charges are invoiced`, which I think is good enough.  But further, I think we could actually strike that entire run and just use "related", per the above.

```suggestion
For all *charges*, Effective Cost reflects all applicable pricing adjustments (e.g., reduced pricing from [*negotiated discounts*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount)). For usage *charges*, Effective Cost includes the consumed portion of *Billed Cost* from related purchase *charges* (e.g., amortized portions of prepayments, drawdowns). For purchase *charges*, Effective Cost excludes related usage *charges* (e.g., usage covered by *commitments*, pre-payments, or marketplace purchases), regardless of when those related *charges* are invoiced.
```

Reviewer 2: Notes from 1/23: Consumed is problematic (in FOCUS, consumed means the amount we actually used - see consumed quantity), decided to go with recognized (doesn't fully capture consumed, but a glossary entry coming in a different work item to explain). Also, adding back the text any 'amounts recognized in related usage charges' or the definition would be wrong. Other updates from Shawn are incorporated.

```suggestion
For all *charges*, Effective Cost reflects all applicable pricing adjustments (e.g., reduced pricing from [*negotiated discounts*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount)). For usage *charges*, Effective Cost includes the recognized portion of *Billed Cost* from related purchase *charges* (e.g., amortized portions of prepayments, drawdowns). For purchase *charges*, Effective Cost excludes any amounts recognized in related usage *charges* (e.g., usage covered by *commitments*, pre-payments, or marketplace purchases which draw down based on usage), regardless of when those related *charges* are invoiced.
```

# Effective Cost

Effective Cost represents the [*amortized*](#glossary:amortization) cost of the [*charge*](#glossary:charge) after applying all reduced rates, discounts, and the applicable portion of relevant, prepaid purchases (one-time or recurring) that covered this *charge*. The *amortized* portion included should be proportional to the [Pricing Quantity](#pricingquantity) and the time granularity of the data. Since amortization breaks down and spreads the cost of a prepaid purchase, to subsequent eligible *charges*, the Effective Cost of the original prepaid *charge* is set to 0. Effective Cost does not mix or "blend" costs across multiple *charges* of the same [*service*](#glossary:service). This cost is denominated in the [Billing Currency](#billingcurrency). The Effective Cost is commonly utilized to track and analyze spending trends.

This column resolves two challenges that are faced by practitioners:

1. Practitioners need to *amortize* relevant purchases, such as upfront fees, throughout the *commitment* and distribute them to the appropriate reporting groups (e.g., [*tags*](#glossary:tag), [*resources*](#glossary:resource)).

> 2. Many [*commitment discount*](#glossary:commitment-discount) constructs include a recurring expense for the *commitment* for every [*billing period*](#glossary:billing-period) and must distribute this cost to the *resources* using the *commitment*. This forces reconciliation between the initial *commitment* [*row*](#glossary:row) per period and the actual usage *rows*.

Author: Resolving @Matt-Cowsert's [original comment](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1678/files/a8a4621b912e940c6a4076437d86a4f5268583e9#r2633425813) and reposting it here for visibility (hidden by a later commit).

> *I don't think we need this section at all. Effective Cost solves more than two challenges for practitioners, and we'll explain those activities in the appendix (vs. the definition).*

Author: Resolving based on the consensus reached during the TF-3 meeting on Jan 09, 2026.

> Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the actual [*resource*](#glossary:resource) or [*service*](#glossary:service) usage or applicable [*contract commitments*](#glossary:contract-commitment) during a given [*charge period*](#glossary:charge-period), regardless of when those costs are invoiced. It is inclusive of all applicable pricing adjustments, such as reduced pricing from [*negotiated*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). This cost is denominated in the [Billing Currency](#billingcurrency). Effective Cost is commonly used for FinOps activities such as reporting, cost forecasting, cost allocation, and chargeback based on actual usage and consumption.

Author: ### OPTION 1' 

*formulated during the Jan 09, 2026 TF-3 meeting*

```suggestion
Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the actual [*resource*](#glossary:resource) or [*service*](#glossary:service) usage or applicable [*contract commitments*]((#glossary:contract-commitment)) during a given [*charge period*](#glossary:charge-period), regardless of when those costs are invoiced. It is inclusive of all applicable pricing adjustments, such as reduced pricing from [*negotiated*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). This cost is denominated in the [Billing Currency](#billingcurrency). Effective Cost is commonly used to support FinOps activites such as accrual-based reporting, forecasting, and cost allocation.
```

Author: resolving in favor of [this comment](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1678#discussion_r2699420884)

> Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the actual [*resource*](#glossary:resource) or [*service*](#glossary:service) usage or applicable [*contract commitments*](#glossary:contract-commitment) during a given [*charge period*](#glossary:charge-period), regardless of when those costs are invoiced. It is inclusive of all applicable pricing adjustments, such as reduced pricing from [*negotiated*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). This cost is denominated in the [Billing Currency](#datasets.costandusage.billingcurrency). Effective Cost is commonly used for FinOps activities such as reporting, cost forecasting, cost allocation, and chargeback based on actual usage and consumption.

Author: ### OPTION 1' 

*formulated during the Jan 09, 2026 TF-3 meeting*

```suggestion
Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the actual [*resource*](#glossary:resource) or [*service*](#glossary:service) usage or applicable [*contract commitments*]((#glossary:contract-commitment)) during a given [*charge period*](#glossary:charge-period), regardless of when those costs are invoiced. It is inclusive of all applicable pricing adjustments, such as reduced pricing from [*negotiated*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). This cost is denominated in the [Billing Currency](#datasets.costandusage.billingcurrency). Effective Cost is commonly used to support FinOps activities such as accrual-based reporting, forecasting, and cost allocation.
```

# Effective Cost

> Effective Cost represents the cost of a [*charge*](#glossary:charge) recognized in the specified [*charge period*](#glossary:charge-period), based on accrual-based accounting principles. Unlike [Billed Cost](#billedcost), which reflects invoiced amounts in a [*billing period*](#glossary:billing-period), Effective Cost reflects the value of usage or entitlements consumed during a *charge period*, regardless of when (or whether) those costs were invoiced. Effective Cost is inclusive of all applicable pricing adjustments, such as reduced unit prices resulting from [*negotiated*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). This cost is denominated in the [Billing Currency](#billingcurrency). Effective Cost is commonly used to support FinOps capabilities, such as accrual-based reporting, cost forecasting, chargebacks, and usage-driven cost allocation.

Reviewer 4: I would recommend using the description "The cost of the *charge* after applying all rate adjustments, discounts, and the applicable portion of relevant, prepaid purchases that covered this *charge*." as the first sentence of this line in place of what's there. 

I would also add a sentence before "This cost is denominated..." that reads "Effective Cost includes cost recognizing the use of applicable prepaid commitments billed in prior charge periods" It might also be useful to add "In public cloud charge records this may include the impact of prepaid reserved instances or savings plans and was typically referred to as amortization of all or partial upfront commitment discounts purchased from cloud providers." if this is not too specific. 

For me, the application of what I've always referred to as amortization is the key aspect of Effective Cost that makes it useful for showing engineering teams the cost of what they have USED this month, vs the cash I will have to pay the cloud provider this month for what they've used. I get that the word Amortization is cloud specific, so happy to move away from it as we incorporate more providers to the mix, but in my opinion this plainer wording speaks to a wider variety of people than the use of cash-based or accrual-based accounting.

Author: Resolved per consensus reached during the TF-3 meeting on Jan 09, 2026.
Update agreed during that meeting and committed on Jan 16, 2026.
@robmartin33 verbally confirmed that the update addresses all concerns raised in this comment.

> Effective Cost represents the cost of a [*charge*](#glossary:charge) recognized in the specified [*charge period*](#glossary:charge-period), based on accrual-based accounting principles. Unlike [Billed Cost](#billedcost), which reflects invoiced amounts in a [*billing period*](#glossary:billing-period), Effective Cost reflects the value of usage or entitlements consumed during a *charge period*, regardless of when (or whether) those costs were invoiced. Effective Cost is inclusive of all applicable pricing adjustments, such as reduced unit prices resulting from [*negotiated*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). This cost is denominated in the [Billing Currency](#billingcurrency). Effective Cost is commonly used to support FinOps capabilities, such as accrual-based reporting, cost forecasting, chargebacks, and usage-driven cost allocation.

Reviewer 5: ```suggestion
Effective Cost represents the cost of a [*charge*](#glossary:charge) recognized in the [*charge period*](#glossary:charge-period) when resources are used or services delivered. Unlike [Billed Cost](#billedcost), it reflects the value of consumption independent of invoicing, inclusive of all  [*negotiated*](#glossary:negotiated-discount) and [*commitment discounts*](#glossary:commitment-discount) terms. This cost is denominated in the [Billing Currency](#billingcurrency). Effective Cost is commonly used to support FinOps activities such as profitability analysis, unit economics, and accrual-based chargeback.
```

Author: One more alternative 😄 and a couple of notes:

- I agree that the term accrual-based accounting is overloaded and should be used carefully. That said, I believe the term “accrual-based” can be used in the context of activities supported by Effective Cost, as long as we clearly define it in the glossary.
- The phrasing “...resource or service usage..." used elsewhere in the spec 
- Profitability analysis and unit economics are not specific to Effective Cost, right?; if we include them here, we should mention them in some other cost metrics as well?

```suggestion
Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the actual [*resource*](#glossary:resource) or [*service*](#glossary:service) usage or entitlements consumption during a given [*charge period*](#glossary:charge-period), regardless of when (or whether) those costs are invoiced. It is inclusive of all applicable pricing adjustments, such as reduced unit prices resulting from [*negotiated*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). This cost is denominated in the [Billing Currency](#billingcurrency). Effective Cost is commonly used for FinOps activities such as reporting, cost forecasting, cost allocation, and chargeback based on actual usage and consumption.
```

Reviewer 6: "regardless of when (or whether) those costs are invoiced" implies to me that the Effective Cost does not need to be eventually consistent with the Billed Cost. I.e. this could be interpreted to add Effective Cost for a free tier usage (which has no Billed Cost), though the subsequent statement about applying adjustments should prevent that.

From my memory, the sum of billed cost should eventually be the same as the sum of effective cost (for the same charge/usage). Do we have an example of where the effective cost would not be reflected in billed cost AND it's a desirable outcome?

Author: @kk09v good catch!

Handling the requirement that the sum of BilledCosts must match the sum of EffectiveCosts is a bit tricky, especially in Marketplace scenarios. Discussion is still needed to decide whether this rule applies at the level of dataset artifacts generated by a single DataGenerator, or globally - particularly relevant in Marketplace scenarios where Purchase charge records come from CSPs, and we expect the corresponding Usage charge records in dataset artifacts from SaaS providers.

Either way, all EffectiveCosts must eventually be invoiced, whether they appear only in SaaS artifacts or in both CSP and SaaS.

Here's an updated version - using:

> ... regardless of when those costs are invoiced...

instead of 

> ...regardless of when (or whether) those costs are invoiced...

```suggestion
Effective Cost represents the cost of a [*charge*](#glossary:charge) based on the actual [*resource*](#glossary:resource) or [*service*](#glossary:service) usage or entitlements consumption during a given [*charge period*](#glossary:charge-period), independent of when that cost is invoiced. It is inclusive of all applicable pricing adjustments, such as reduced unit prices resulting from [*negotiated*](#glossary:negotiated-discount) or [*commitment discounts*](#glossary:commitment-discount). This cost is denominated in the [Billing Currency](#billingcurrency). Effective Cost is commonly used for FinOps activities such as reporting, cost forecasting, cost allocation, and chargeback based on actual usage and consumption.
```

Author: Resolving per consensus reached during the TF-3 meeting on Jan 09, 2026. 
Update agreed during that meeting and committed on Jan 16, 2026.

> This column resolves two challenges that are faced by practitioners:

Reviewer 5: I don't think we need this section at all. Effective Cost solves more than two challenges for practitioners, and we'll explain those activities in the appendix (vs. the definition).

Author: Resolving and [reposting here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1678/files#r2673934491) for visibility (as it was hidden by a subsequent commit).

## Description

The *amortized* cost of the *charge* after applying all reduced rates, discounts, and the applicable portion of relevant, prepaid purchases (one-time or recurring) that covered this *charge*.

### Concerning Granularity and Distribution of Recurring Fee

Service providers should distribute the *commitment* purchase amount instead of including a *row* at the beginning of a period so practitioners do not need to manually distribute the fee themselves.

### Concerning Amortization Approaches

> Eligible purchases should be *amortized* using a methodology determined by the service provider that reflects the needs of their customer base and is proportional to the Pricing Quantity and the time granularity of the *row*. Should a practitioner desire to *amortize* relevant purchases using a different approach, the practitioner can do so using the [Billed Cost](#billedcost) for the line item representing the initial purchase.

Author: I suggest removing these two sections, or at least revisiting them, as they are not sufficient in their current form.

---

**Reasoning:**

- The current heading level implies they are part of the Description. They might be better addressed in the first section, or we should introduce a level-2 heading above them (if we decide to keep them within the EffectiveCost definition at all).

- These sections effectively recommend how BilledCost and EffectiveCost for commitment discount scenarios should be provided, but they assume a lot, for example: 
   - that it is clear what an eligible purchase represents
   - that the reader already understands how providers currently expose these records
 
- We should probably **add more context around different commitment payment models and scenarios** (No-Upfront, All-Upfront, Partial-Upfront) and move the detailed discussion of these specific scenarios to an **appendix** or possibly even an **attribute** and specify there what we expect or recommend for **both BilledCost and EffectiveCost**

- Instead of Recurring Fee (AWS-specific terminology), we may want to use more generic wording such as commitment discount purchase records

- We may consider focusing on cost distribution instead of amortization

***Note:** Upcoming work in FR #1459 (Extend support for Contract Commitment dataset) or FR #1070 (Clarify commitment concepts and glossary entries) may introduce Payment Model as a column or glossary entry, which could then be referenced when refining these recommendations.*

Author: Resolving based on the consensus reached during the TF-3 meeting on Jan 09, 2026.

## Description

> Cost of a *charge* based on the *resources* or *services* used or *contract commitments* consumed in a given *charge period*.

Reviewer 2: ```suggestion
Cost of a *charge* based on the *resources* used, *services* used, or *contract commitments* consumed in a given *charge period*.
```

> Cost of a *charge* based on the *resources* used, *services* used, or *contract commitments* consumed in a given *charge period*.

Reviewer 2: ```suggestion
Cost of a *charge* based on the *resources* used, *services* used, or *contract commitments* recognized in a given *charge period*.
```

## Description

> Cost of a *charge* recognized in the specified *charge period*, based on accrual-based accounting principles.

Reviewer 4: per previous comment, I'd recommend moving away from talking about accrual-based accounting, and instead revert to the previous description excluding the word "amortized" which could be similarly misunderstood outside of the cloud world.

Author: An alternative that avoids using the term accrual-based accounting. 

*Notes:*
- *As a general rule, we use the first sentence of the preamble in the descriptions.*
- *In previous discussions, the team agreed that the first sentence should focus on what Effective Cost represents, rather than how it is calculated or what it includes or excludes (for example, reduced prices or discounts).*
- *That said, we may want to consider adding a second sentence to the Effective Cost and Billed Cost descriptions to explicitly reference reduced unit prices.*

```suggestion
Cost of a *charge* based on the actual *resource* or *service* usage or entitlements consumption during a given *charge period*, independent of when that cost is invoiced.
```

Author: Resolving per consensus reached during the TF-3 meeting on Jan 09, 2026. 
Update agreed during that meeting and committed on Jan 16, 2026.
@robmartin33 verbally confirmed that the update addresses all concerns raised in this comment.
